File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int i = 0, j;
        for(j = 0; j < nums.length; ++j) {
            if(nums[j] == 0) --k;
            // move i if k < 0 and increase k only if you see a 0 at i
            if(k < 0 && nums[i++] == 0) ++k;
        }
        return j - i;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        i = j = 0
        for j in range (0, len(nums)):
            if nums[j] == 0: k -= 1
            if k < 0 and nums[i] == 1: 
                i += 1
            elif k < 0 and nums[i] == 0: 
                i += 1
                k += 1
        return j - i + 1